src/scanner.c

```diff
@@ -258,6 +259,38 @@ bool is_valid_string_delim(char c) {
   return c == '\'' || c == '"';
 }
 
+// Symbols allowed as pattern delimiter
+bool is_valid_syn_pattern_delimiter(char c) {
+  static char valids[] = {
```

Comment:
Maybe this can be made a const to be in the ro section ?
```suggestion
  static const char valids[] = {
```